ckanext/dia/harvester/dcat.py

```diff
@@ -41,7 +44,7 @@ def _clean_email(self, email):
     def _clean_spatial(self, spatial):
         # Convert things like "173.0039,-42.3167,174.2099,-41.0717" to
         # Polygon using templates from CSW
-        if isinstance(spatial, six.string_types):
+        if isinstance(spatial, six.string_types) and spatial != "":
```

Comment:
Almost wonder if we should test the structure a little better, even a regex that checks that there are are only digits, full stops, minus signs and commas would be useful. Do we have a way of sending custom error messages in these scenarios?

Reply:
`^[\d\-,\.]+$` should work fine to test that the string only contains the valid/expected characters

Reply:
Updated. I have tightened the regex and also added a check for some other invalid data we were receiving.